akka-http-core/src/main/scala/akka/http/impl/engine/HttpConnectionTimeoutException.scala

```diff
@@ -0,0 +1,5 @@
+package akka.http.impl.engine
+
+import scala.util.control.NoStackTrace
+
+class HttpConnectionTimeoutException(msg: String) extends RuntimeException(msg) with NoStackTrace
```

Comment:
very good about the with NoStackTrace btw.
I wonder if it could be made an object even hmm